grading/main_runner.cpp

```diff
@@ -36,31 +41,52 @@ int main(int argc, char *argv[]) {
   // LOAD HW CONFIGURATION JSON
   nlohmann::json config_json = LoadAndProcessConfigJSON(rcsid);
 
-  nlohmann::json grading_parameters = config_json.value("grading_parameters",nlohmann::json::object());
-  int AUTO_POINTS         = grading_parameters.value("AUTO_POINTS",0);
-  int EXTRA_CREDIT_POINTS = grading_parameters.value("EXTRA_CREDIT_POINTS",0);
-  int TA_POINTS           = grading_parameters.value("TA_POINTS",0);
-  int TOTAL_POINTS        = grading_parameters.value("TOTAL_POINTS",AUTO_POINTS+TA_POINTS);
+  // nlohmann::json grading_parameters = config_json.value("grading_parameters",nlohmann::json::object());
+  // int AUTO_POINTS         = grading_parameters.value("AUTO_POINTS",0);
+  // int EXTRA_CREDIT_POINTS = grading_parameters.value("EXTRA_CREDIT_POINTS",0);
+  // int TA_POINTS           = grading_parameters.value("TA_POINTS",0);
+  // int TOTAL_POINTS        = grading_parameters.value("TOTAL_POINTS",AUTO_POINTS+TA_POINTS);
 
   // necessary since the untrusted user does not have a home directory
   setenv("DYNAMORIO_CONFIGDIR", ".", 1);
 
   system("find . -type f -exec ls -sh {} +");
 
   // Run each test case and create output files
+  std::vector<std::string> required_capabilities = stringOrArrayOfStrings(config_json, "required_capabilities");
+  
+  bool windowed = false;
+  if (std::find(required_capabilities.begin(), required_capabilities.end(), "windowed") != required_capabilities.end()){
+    windowed = true;
+  }
+
   nlohmann::json::iterator tc = config_json.find("testcases");
   assert (tc != config_json.end());
+
+  if(test_case_to_run != -1){
+    assert (test_case_to_run < tc->size());
+  }else{
+    std::cout << "Running all testcases in a single run." << std::endl;
+  }
+
   for (unsigned int i = 0; i < tc->size(); i++) {
 
+    TestCase my_testcase(config_json,i);
+
+    if (my_testcase.isFileCheck() || my_testcase.isCompilation()){
+      continue;
+    }
+
+    if(test_case_to_run != -1 &&  test_case_to_run != i){
+      continue;
+    }
```

Comment:
This continue should be above the instantiation of the my_testcase object.

Reply:
This is a necessary workaround. When test cases are instantiated, their “number” is set via a static global int, so the first one created is 0, then 1, 2, 3, etc. To avoid tearing apart the test case class, we decided to just instantiate unused test cases for now.